### PR TITLE
MESSANGER-31

### PR DESCRIPTION
**Pull request by Nikita8Sannikov**
https://tracker.yandex.ru/MESSANGER-31
- [x] improve number validation

### CODE REVIEW

File: app/src/modules/SignUp/Form.tsx [hunk 1]
```diff
@@ -2,204 +2,260 @@ import React from "react";
...
+			: REQUIRED_FIELD;
+	};
 
-                        onChange={e => onChange("email", e.target.value, validateEmail)}
-                    />
-                </Grid>
+	const validatePhone = (value: string, country: CountryData) => {
+		if (value) {
+			const numbersCount = value.match(/\d/g);
+			const numberLength = (country as CountryData).format.match(
```
Comment (makebelieve-sk) [inline app/src/modules/SignUp/Form.tsx:75]: "country" already have the type CountryData. Why you use "as CountryData"?

File: app/src/modules/SignUp/Form.tsx [hunk 2]
```diff
@@ -2,204 +2,260 @@ import React from "react";
...
+				/\./g
+			)?.length;
+			return numbersCount &&
+				numbersCount.length &&
+				numbersCount.length !== numberLength
+				? `Длина номера телефона должна быть ${numberLength} символов`
```
Comment (makebelieve-sk) [inline app/src/modules/SignUp/Form.tsx:81]: whitespaces!!!

File: app/src/modules/SignUp/Form.tsx (hunk 1 above)
Comment (Nikita8Sannikov) [reply app/src/modules/SignUp/Form.tsx:75]: You're right! Thanks

File: app/src/modules/SignUp/Form.tsx (hunk 2 above)
Comment (Nikita8Sannikov) [reply app/src/modules/SignUp/Form.tsx:81]: Are you talkink about prettier setting or what?

File: app/src/modules/SignUp/Form.tsx [hunk 3]
```diff
@@ -2,204 +2,258 @@ import React from "react";
...
+			: REQUIRED_FIELD;
+	};
 
-                        error={Boolean(formValues.errors.email)}
-                        helperText={formValues.errors.email ? formValues.errors.email : null}
-                        value={formValues.values.email}
+	const validateEmail = (value: string) => {
+		return value
+			? value.match(
+					/^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
Comment (makebelieve-sk) [inline app/src/modules/SignUp/Form.tsx:65]: Maybe we can create a separate function to check the correct email address in utils.ts?

File: app/src/modules/SignUp/Form.tsx [hunk 4]
```diff
@@ -2,204 +2,257 @@ import React from "react";
...
 import TextField from "@mui/material/TextField";
-import PhoneInput from "react-phone-input-2";
+import PhoneInput, { CountryData } from "react-phone-input-2";
 
 import { REQUIRED_FIELD } from "../../utils/constants";
 import { ISignUpState } from "../../pages/SignUp";
-
 import "./sign-up.scss";
 import "react-phone-input-2/lib/material.css";
+import { emailCheck } from "../../utils/email-check";
```
Comment (makebelieve-sk) [inline app/src/modules/SignUp/Form.tsx:11]: one moment about imports
i will write you in tg

**Pull request merged**